src/Native/System.Native/pal_networking.cpp

```diff
@@ -2013,3 +2018,21 @@ extern "C" Error WaitForSocketEvents(int32_t port, SocketEvent* buffer, int32_t*
 
     return WaitForSocketEventsInner(port, buffer, count);
 }
+
+extern "C" int32_t PlatformSupportsMultipleConnectAttempts()
+{
+#if defined(PLATFORM_SUPPORTS_MULTIPLE_CONNECT_ATTEMPTS)
```

Comment:
I just checked the code, we are explicitly passing it via compiler flag in case of Linux. Can we detect this feature in `configure.cmake` somehow? It will then be easier to detect the capability on future platforms as well. Perhaps even some linux might not have this capability (Alpine or some other odd distro).